onnxruntime/core/graph/graph.cc

```diff
@@ -3498,13 +3562,45 @@ common::Status Graph::LoadFromOrtFormat(const onnxruntime::experimental::fbs::Gr
 
   // Initializers
   auto fbs_initializers = fbs_graph.initializers();
+  auto fbs_sparse_initializers = fbs_graph.sparse_initializers();
+  flatbuffers::uoffset_t map_size = (fbs_initializers != nullptr ? fbs_initializers->size() : 0U) +
+                                    (fbs_sparse_initializers != nullptr ? fbs_sparse_initializers->size() : 0U);
+
+  if (map_size > 0) {
+    name_to_initial_tensor_.reserve(map_size);
+  }
+
   if (fbs_initializers) {
-    name_to_initial_tensor_.reserve(fbs_initializers->size());
     for (const auto* fbs_tensor : *fbs_initializers) {
       ORT_RETURN_IF(nullptr == fbs_tensor, "Initializer tensor is missing. Invalid ORT format model.");
       TensorProto* initializer = deserialized_proto_data_.add_initializer();
       ORT_RETURN_IF_ERROR(experimental::utils::LoadInitializerOrtFormat(*fbs_tensor, *initializer));
-      name_to_initial_tensor_[initializer->name()] = initializer;
+      auto p = name_to_initial_tensor_.emplace(initializer->name(), initializer);
+      if (!p.second) {
+        LOGS(logger_, WARNING) << "Duplicate initializer (dense or ConstantNode): '" << initializer->name()
+                              << "' the model will use the latest encountered initializer"
+                              << ". Please, fix your model.";
+        p.first->second = initializer;
+      }
+    }
```

Comment:
technically this can never happen given we throw away the duplicate when loading the onnx format model (which we create the ort format model from)